packages/angular/src/tracing.ts

```diff
@@ -241,3 +279,44 @@ export function TraceMethodDecorator(): MethodDecorator {
     return descriptor;
   };
 }
+
+/**
+ * Recursively traverses the routerstate and its children to collect a map of parameters on the entire route.
+ *
+ * Because Angular supports child routes (e.g. when creating nested routes or lazy loaded routes), we have
+ * to not only visit the root router snapshot but also its children to get all parameters of the entire route.
+ *
+ * @param activatedRouteSnapshot the root router snapshot
+ * @returns a map of params, mapping from a key to an array of values
+ */
+export function getParamsOfRoute(activatedRouteSnapshot: ActivatedRouteSnapshot): ParamMap {
+  function getParamsOfRouteH(routeSnapshot: ActivatedRouteSnapshot, accParams: ParamMap): ParamMap {
```

Comment:
I don't understand why this is a tree and not just a chain of route snapshots. A tree would imply that there are multiple routes active (leaf nodes of the tree), can that be the case? Multi-layouts? Do we consider those as "navigation events" where we want to start transactions?

Reply:
So my best guess is that we're not going to encounter multi-child routes, which is why for the new approach we're essentially treating the route snapshot as a chain instead of a tree. My assumption: The `ResolveEnd` event holds the router state of the newly resolved (and soon to be activated) route and not the entire router state. 
If we get reports of wrong transaction names, we can always revisit this strategy. Although I'd argue that in this case we'd need to rethink how we treat such navigations (i.e. how the transaction name should look like then).